internal/connector/connector.go

```diff
@@ -481,31 +481,42 @@ func Run(ctx context.Context, options Options) error {
 		endpoint := fmt.Sprintf("%s:%d", host, port)
 		logging.S.Debugf("connector serving on %s", endpoint)
 
-		if destination.ID == 0 {
-			destination.Connection.CA = string(caCertPEM)
-			destination.Connection.URL = endpoint
+		namespaces, err := k8s.Namespaces()
+		if err != nil {
+			logging.S.Errorf("could not get kubernetes namespaces: %w", err)
+			return
+		}
 
+		clusterRoles, err := k8s.ClusterRoles()
+		if err != nil {
+			logging.S.Errorf("could not get kubernetes cluster-roles: %w", err)
+			return
+		}
+
+		switch {
+		case destination.ID == 0:
 			isClusterIP, err := k8s.IsServiceTypeClusterIP()
 			if err != nil {
-				logging.S.Debugf("could not check destination service type: %v", err)
+				logging.S.Debugf("could not determine service type: %v", err)
 			}
 
 			if isClusterIP {
-				logging.S.Warn("registering with cluster IP, it may not be externally accessible without an ingress or load balancer")
+				logging.S.Warn("registering Kubernetes connector with ClusterIP. it may not be externally accessible. if you are experiencing connectivity issues, consider switching to LoadBalancer or Ingress")
 			}
 
-			err = createDestination(client, destination)
-			if err != nil {
-				logging.S.Errorf("initializing destination: %v", err)
-				return
-			}
-		} else if destination.Connection.URL != endpoint || destination.Connection.CA != string(caCertPEM) {
+			fallthrough
+
+		case !slicesEqual(destination.Resources, namespaces), !slicesEqual(destination.Roles, clusterRoles):
```

Comment:
This `switch` control flow is confusing: could we just use regular `if` statements ?

E.g.
```
if destination.ID == 0 {
  // create a destination
} else {
 // update the destination if fields have changed
}
```

Not urgent: but this goroutine is accessing an unprotected variable `destination`. While in practice it shouldn't be an issue we should move it in this goroutine

Reply:
The switch allows more concise code. It's not straight forward since multiple things are checked and updated before calling the API. `destination` is only after modified in the goroutine after being initialized. It's important its available for all goroutines as it caches the destination object which is used to determine when to call create or update

Reply:
Created #1942